src/arkode/arkode_mristep.c

```diff
@@ -1082,42 +1078,65 @@ int mriStep_Init(void* arkode_mem, int init_type)
                       "Error in coupling table");
       return (ARK_ILL_INPUT);
     }
-
+    
     /* stage types */
-    if (step_mem->stagetypes)
+    step_mem->stagetypes = realloc(step_mem->stagetypes,
+                                   step_mem->stages * sizeof(*step_mem->stagetypes));
```

Comment:
I'm okay with changing to `realloc`, but (a) please cast the result to the correct type (`int*`), and (b) `memset` the new memory to all zeros to retain `calloc`'s initialization feature.